apache/superset
# fix(pvt2): migrations from legacy pivot table error when form_data have pieces of pvt2

### SUMMARY
Hello! 
Migration from legacy pivot table v2 sometimes returns an error due to pieces of  Pivot Table V2 form_data in the legacy pivot. This PR fixes it. Pieces of form_data can occur when users switch from pvt2 tables to legacy pivot.


### TESTING INSTRUCTIONS
1. Create an save Pivot Table v2 visualization
2. Switch to the legacy pivot table and save
3. Try to migrate and see the error "Dublicate key".

Or just try to migrate form_data that is already generated by me and see the error.
```python
{'adhoc_filters': [], 'aggregateFunction': 'Sum', 'applied_time_extras': {}, 'colOrder': 'key_a_to_z', 'colTotals': True, 'columns': ['date_obr'], 'conditional_formatting': [], 'dashboards': [], 'datasource': '2858__table', 'date_format': 'smart_date', 'granularity_sqla': 'date_', 'groupby': ['Name_gr'], 'groupbyColumns': ['date_obr'], 'groupbyRows': ['id_group', 'TT'], 'label_colors': {}, 'metrics': ['count'], 'metricsLayout': 'COLUMNS', 'number_format': 'SMART_NUMBER', 'order_desc': True, 'pandas_aggfunc': 'Sum', 'pivot_margins': True, 'rowOrder': 'value_z_to_a', 'rowTotals': True, 'row_limit': 1000, 'series_limit_metric': ['count'], 'shared_label_colors': {}, 'slice_id': 16619, 'time_grain_sqla': 'P1D', 'time_range': 'Last week', 'timeseries_limit_metric': 'count', 'valueFormat': ',d', 'viz_type': 'pivot_table_v2', 'extra_filters': [], 'dataMask': {}, 'extraControls': {}}
```
### ADDITIONAL INFORMATION
